EIPS/eip-3074.md

```diff
@@ -56,39 +53,66 @@ The variable has the same scope as the program counter -- `authorized` persists
 
 ### `AUTH` (`0xf6`)
 
-A new opcode `AUTH` shall be created at `0xf6`. It shall take four stack element inputs and returns one stack element.
+A new opcode `AUTH` shall be created at `0xf6`. It shall take four stack element inputs (the last two describing a memory range), and it shall return one stack element.
 
 #### Input
 
+##### Stack
+
 | Stack      | Value        |
 | ---------- | ------------ |
 | `top - 0`  | `commit`     |
-| `top - 1`  | `yParity`    |
-| `top - 2`  | `r`          |
-| `top - 3`  | `s`          |
+| `top - 1`  | `authority`  |
+| `top - 2`  | `offset`     |
+| `top - 3`  | `length`     |
+
+##### Memory
+
+The final two stack arguments (`offset` and `length`) describe a range of memory. The format of the contents of that range is:
```

Comment:
The rationale for this is that it is now possible to also support other signature types, right? If that is the case, how should these be distinguished? It is clear that `memory[offset : offset+32]` is either all zeros or 31 zero bytes and 1 byte representing `1`, so it would be possible to do some versioning there to signal whatever signature type is used. I am not sure if this versioning should be baked in from the start, though :thinking:

Reply:
> If that is the case, how should these be distinguished?

The target address will eventually know how to validate its own signature format.

Reply:
Ah right, this makes sense.